src/drivers/hermes/hermes.c

```diff
@@ -21,28 +21,153 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
+#include <linux/cdev.h>
 #include <linux/module.h>
 #include <linux/pci.h>
 
 #define PCI_VENDOR_EIDETICOM	0x1de5
 #define PCI_HERMES_DEVICE_ID	0x3000
 
+#define HERMES_MINOR_BASE	0
+#define HERMES_MINOR_COUNT	16
+#define HERMES_NAME		"hermes"
+
 extern int xdma_probe_one(struct pci_dev *pdev, const struct pci_device_id *id);
 extern void xdma_remove_one(struct pci_dev *pdev);
+extern void *xdma_pci_dev_get_priv(void *xpdev);
+extern void *xdma_pci_dev_set_priv(void *xpdev, void *x);
+
+static struct class *hermes_class;
+DEFINE_IDA(hermes_ida);
+static dev_t hermes_devt;
+
+struct hermes_dev {
+	struct device dev;
+	struct pci_dev *pdev;
+	struct cdev cdev;
+	int id;
+};
 
 static struct pci_device_id pci_ids[] = {
 	{ PCI_DEVICE(PCI_VENDOR_EIDETICOM, PCI_HERMES_DEVICE_ID), },
 	{ 0, }
 };
 MODULE_DEVICE_TABLE(pci, pci_ids);
 
+static ssize_t hermes_read(struct file *filp, char __user *buf, size_t len,
+		    loff_t *off)
+{
+	return 0;
+}
+
+static ssize_t hermes_write(struct file *filp, const char __user *buf,
+		size_t len, loff_t *off)
+{
+	return len;
```

Comment:
Better to not introduce these emtpy stubs it's confusing to the reviewer.... Just add them to the fops when you implement the function.

Looks like hermes_read is never actually used.